osm.c: ikey is returned as int

Don't cast to char and back without a reason.

--- a/osm.c
+++ b/osm.c
@@ -443,7 +443,7 @@ osm_features_init(void)
 }
 
 
-static char
+static int
 osm_feature_ikey(const char *key)
 {
 	int result;
@@ -535,7 +535,7 @@ osm_node_tag(const char *args, const cha
 	const char **avp = &attrv[0];
 	const char *key = "", *value = "";
 	char *str;
-	char ikey;
+	int ikey;
 	
         while (*avp) {
 		if (strcmp(avp[0], "k") == 0)
